pkg/sync/rpc/list.go

```diff
@@ -42,13 +52,24 @@ func (b *BackupList) BackupAdd(backupID string, backup *replica.BackupStatus) er
 		return fmt.Errorf("empty backupID")
 	}
 
+	// If a record already exists, remove it before adding the new one.
+	// Ignore errors - we don't really expect to find it, normally.
+	_ = b.BackupDelete(backupID)
+
 	b.Lock()
 	b.infos = append(b.infos, &BackupInfo{
 		backupID:     backupID,
 		backupStatus: backup,
 	})
 	b.Unlock()
 
+	if backupListHighWaterMark < len(b.infos) {
+		backupListHighWaterMark = len(b.infos)
+		if 0 == backupListHighWaterMark%BackupListHighWaterMarkQuantum {
```

Comment:
I'm not sure I understand this. This only returns `true` if `backupListHighWaterMark` is a multiple of 10, right? Are we trying to limit how often we log (only once every ten increases)?

Reply:
Yep.  If we do have 1000 backups (possible) we don't really want to log 1000 times, but we don want to know if we've got a runaway list.  So this is just a simple-minded throttle.

Reply:
Got it. Is there any known way to have 1000 backups after these changes? Shouldn't we delete any more than 10 error and 5 completed in `refresh` on 73?

(Not saying I have a problem with the throttle. Just making sure I understand our expectations.)

Reply:
It is certainly possible to have 1000 backups in progress - not yet either complete or error.  The refresh function only prunes those backups that have reached a conclusion.